superset/config.py

```diff
@@ -351,8 +351,9 @@ def _try_json_readsha(filepath, length):  # pylint: disable=unused-argument
 SUPERSET_WEBSERVER_DOMAINS = None
 
 # Allowed format types for upload on Database view
-# TODO: Add processing of other spreadsheet formats (xls, xlsx etc)
-ALLOWED_EXTENSIONS = {"csv", "tsv"}
+EXCEL_EXTENSION = ["xlsx", "xls"]
+CSV_EXTENSION = ["csv", "tsv"]
+ALLOWED_EXTENSIONS = {"csv", "tsv", "xlsx", "xls"}
```

Comment:
```suggestion
EXCEL_EXTENSION = {"xlsx", "xls"}
CSV_EXTENSION = {"csv", "tsv"}
ALLOWED_EXTENSIONS = {*EXCEL_EXTENSION, *CSV_EXTENSION}
```
The first two should be sets, and the last one should be a union of the above.